### PR TITLE
Add generate options

### PR DESCRIPTION
**Pull request by tonybase**
BodyStringify: Requests are controlled by Content-type

### CODE REVIEW

File: internal/plugin/generate.go [hunk 1]
```diff
@@ -13,7 +13,18 @@ import (
...
-func Generate(request *pluginpb.CodeGeneratorRequest) (*pluginpb.CodeGeneratorResponse, error) {
+// Options is generate options.
+type Options struct {
+	// UseProtoNames uses proto field name instead of lowerCamelCase name in JSON
+	// field names.
+	UseProtoNames bool
+	// UseEnumNumbers emits enum values as numbers.
+	UseEnumNumbers bool
+	// BodyStringify JSON.stringify body.
+	BodyStringify bool
```
[the remark this answers is not in the document]
Comment (ericwenn) [reply internal/plugin/generate.go:24]: If this is disabled, ie no JSON.stringify, what format will the body have?
Even if not using JSON fields, I think the body should still be stringified as JSON, but I could be wrong.

**Review by ericwenn — COMMENTED** [review]
Thanks for the contribution. This sounds reasonable.
Out of interest, how are you exposing your API such that you prefer proto names instead of JSON names?

File: internal/plugin/generate.go [hunk 2]
```diff
@@ -13,7 +13,18 @@ import (
...
 )
 
-func Generate(request *pluginpb.CodeGeneratorRequest) (*pluginpb.CodeGeneratorResponse, error) {
+// Options is generate options.
+type Options struct {
+	// UseProtoNames uses proto field name instead of lowerCamelCase name in JSON
+	// field names.
+	UseProtoNames bool
+	// UseEnumNumbers emits enum values as numbers.
+	UseEnumNumbers bool
```
[the remark this answers is not in the document]
Comment (tonybase) [reply internal/plugin/generate.go:23]: Now we're referring to protojson: https://pkg.go.dev/google.golang.org/protobuf@v1.28.1/encoding/protojson#MarshalOptions

File: internal/plugin/generate.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tonybase) [reply internal/plugin/generate.go]: Our body actually supports form or json encoding, which we can specify through Options.

File: internal/plugin/generate.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ericwenn) [reply internal/plugin/generate.go:23]: What I mean is, if i specify `UseEnumNumbers` the enum types that are generated in typescript will not be usable any more (it is strings). Should these still be generated if this option is set?

File: internal/plugin/generate.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ericwenn) [reply internal/plugin/generate.go]: I did not intend for my first comment to be the documentation for the field.
```
DisableBodyStringify disables encoding the body with `JSON.stringify` before being passed to the handler.
```

**Comment by ericwenn** [comment]
@tonybase please take a look at the failing CI.
Regarding commit messages we enforce conventional commits in order to automatically tag new versions.
I suggest squashing all commits and prefixing with `feat:`

**Pull request closed without merging**